__tests__/InlineHtmlStripStylesTransformer.test.ts

```diff
@@ -0,0 +1,214 @@
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ * 
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ * 
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ * 
+ */
+
+import * as tsc from 'typescript'
+import * as transformer from '../InlineHtmlStripStylesTransformer'
+
+const CODE_WITH_TEMPLATE_URL = `
+import { Component } from '@angular/core';
+
+@Component({
+  templateUrl: './page.html'
+})
+export class AngularComponent {
+}
+`
+
+const CODE_WITH_STYLE_URLS = `
+import { Component } from '@angular/core';
+
+@Component({
+  styleUrls: [
+    './fancy-styles.css',
+    './basic-styles.scss'
+  ]
+})
+export class AngularComponent {
+}
+`
+
+const CODE_WITH_STYLES = `
+import { Component } from '@angular/core';
+
+@Component({
+  styles: [
+    'body: { display: none }',
+    'html: { background-color: red }'
+  ]
+})
+export class AngularComponent {
+}
+`
+
+const CODE_WITH_ALL_DECORATOR_PROPERTIES = `
+import { Component } from '@angular/core';
+
+@SomeDecorator({
+  value: 'test'
+})
+@Component({
+  templateUrl: './page.html',
+  styleUrls: [
+    './fancy-styles.css',
+    './basic-styles.scss'
+  ],
+  styles: [
+    'body: { display: none }',
+    'html: { background-color: red }'
+  ],
+  unaffectedProperty: 'whatever'
+})
+@SomeOtherDecorator({
+  prop: 'ok'
+})
+export class AngularComponent {
+}
+`
+const createFactory = () => {
+  return transformer.factory({ compilerModule: tsc } as any)
+}
+const transpile = (source: string) => tsc.transpileModule(source, { transformers: { before: [createFactory()] } })
+
+describe('inlining template and stripping styles', () => {
+  it('should have correct signature', () => {
+    expect(transformer.name).toBe('angular-component-inline-template-strip-styles')
+    expect(typeof transformer.version).toBe('number')
+    expect(transformer.version).toBeGreaterThan(0)
+    expect(typeof transformer.factory).toBe('function')
+  })
+
+  it('should strip styleUrl assignment', () => {
+    const out = transpile(CODE_WITH_STYLE_URLS)
+
+    const compiledBeginningOfComponent = 'var AngularComponent'
+
+    const output = out.outputText
+    expect(output).toContain(compiledBeginningOfComponent)
+
+    // just compare class part, generated decorator function is not interesting
+    const classPart = output.substr(output.indexOf(compiledBeginningOfComponent))
+    expect(classPart).toEqual(
```

Comment:
I think it would make sense to snapshot the output of `classPart` and provide a descriptive name of what's expected to be inside this output

Reply:
:+1: for using snapshots.

I would remove the substringing and have the decorator-bootstrapping in the snapshot as well, it does not distract as much in there.